client/mysqlbinlog.cc

```diff
@@ -2543,8 +2547,7 @@ static Exit_status dump_local_log_entries(PRINT_EVENT_INFO *print_event_info,
     if (mysqlbinlog_file_reader.event_data_istream()->is_binlog_encrypted() &&
         mysqlbinlog_file_reader.get_error_type() !=
             Binlog_read_error::READ_EOF &&
-        (!ev || ev->get_type_code() != binary_log::START_ENCRYPTION_EVENT)) {
-      DBUG_ASSERT(ev == nullptr);
+        !ev) {
```

Comment:
Check clang-format before pushing, looks suspicious

Reply:
I have checked it, seems OK.